module/TheWitcherTRPG.js

```diff
@@ -281,3 +282,46 @@ Handlebars.registerHelper("window", function (...props) {
 Handlebars.registerHelper("includes", function (csv, substr) {
     return csv.split(",").map(v => v.trim()).includes(substr);
 });
+
+
+/* -------------------------------------------- */
+/*  Combat Hooks                                */
+/* -------------------------------------------- */
+Hooks.on("updateCombat", async (combat, turn, data, options) => {
+    console.log("updateCombat hook triggered!");
+
+    // Get the current combatant based on the current turn index
+    const currentTurnIndex = combat.turn;
+    const currentCombatant = combat.turns[currentTurnIndex];
+
+    if (!currentCombatant) return;
+
+    const actor = currentCombatant.actor;
+
+    if (!actor) return;
+
+    // Check if the actor has any effects
+    if (actor.effects.size > 0) {
+        actor.effects.forEach(async (effect) => {
+            switch (effect.name) {
```

Comment:
I would like if you could change it in a way that it checks the effect for certain fields and acts on these fields

Reply:
What fields do you mean? Do you have an example?

Reply:
see https://github.com/witchertrpg-foundryvtt/TheWitcherTRPG/pull/138#discussion_r1733199334